src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -432,9 +439,17 @@ private void VerifyConnected()
 				foreach (var ipAddress in ipAddresses)
 				{
 					TcpClient tcpClient = null;
+					int originalSendTimeout, originalReceiveTimeout;
 					try
 					{
 						tcpClient = new TcpClient(ipAddress.AddressFamily);
+
+						// unix platforms block on connect, use send/receive timeouts: https://github.com/dotnet/corefx/issues/20954
+						originalSendTimeout = tcpClient.Client.SendTimeout;
+						originalReceiveTimeout = tcpClient.Client.ReceiveTimeout;
+						tcpClient.Client.SendTimeout = cs.ConnectionTimeoutMilliseconds;
```

Comment:
I think get/set socket timeouts call `getsocketopt` and `setsocketopt` syscalls. This commit adds 6 get/set socketopt syscalls. These need to be restricted to non-windows platforms as to not cause unnecessary syscalls.